cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/AbstractMapReduceContextBuilder.java

```diff
@@ -109,8 +112,8 @@ public BasicMapReduceContext build(MapReduceMetrics.TaskType type,
     MapReduceSpecification spec = program.getApplicationSpecification().getMapReduce().get(program.getName());
     BasicMapReduceContext context =
       new BasicMapReduceContext(program, type, RunIds.fromString(runId), taskId, runtimeArguments, datasets, spec,
-                                logicalStartTime, workflowBatch, discoveryServiceClient, metricsCollectionService,
-                                datasetFramework, adapterSpec, pluginInstantiator);
+                                logicalStartTime, workflowBatch, workflowToken, discoveryServiceClient,
```

Comment:
`workflowBatch` is the part of existing code. We pass the name of the program in it, so that we can correctly initialize the `Program` from its specifications. It won't be the unique name of the node.